File: source/Emphasize/Format Definitions/ItalicCode.vb
Imports Microsoft.VisualStudio.Text.Classification
Imports Microsoft.VisualStudio.Utilities
Imports System.ComponentModel.Composition


Namespace FormatDefinitions

    <ClassificationType(ClassificationTypeNames:=ItalicCode.Name)>
    <Export(GetType(EditorFormatDefinition))>
    <Name(ItalicCode.Name)>
    <Order(After:=Priority.High)>
    <UserVisible(True)>
    Public NotInheritable Class ItalicCode
        Inherits ClassificationFormatDefinition


        Public Const Name As String = "Comment - Italic, Code Span"


        <Export(GetType(ClassificationTypeDefinition))>
        <BaseDefinition("Comment")>
        <Name(Name)>
        Public Shared ReadOnly ClassificationType As ClassificationTypeDefinition


        Public Sub New()
            DisplayName = Name
            IsItalic = True
            BackgroundBrush = Code.CreateBackgroundBrush()
            BackgroundOpacity = Code.GetBackgroundOpacity()
        End Sub

    End Class

End Namespace

File: source/Emphasize/FormatDefinitions/ItalicCode.cs
#nullable enable

using Microsoft.VisualStudio.Text.Classification;
using Microsoft.VisualStudio.Utilities;
using System.ComponentModel.Composition;


namespace Emphasize.FormatDefinitions {

    [ClassificationType(ClassificationTypeNames = Name)]
    [Export(typeof(EditorFormatDefinition))]
    [Name(Name)]
    [Order(After = Priority.High)]
    [UserVisible(true)]
    public class ItalicCode : ClassificationFormatDefinition {

        public const string Name = "Emphasize - Italic, Code";


        [Export(typeof(ClassificationTypeDefinition))]
        [Name(Name)]
        public static readonly ClassificationTypeDefinition? ClassificationType;


        public ItalicCode() {
            DisplayName = Name;
            IsItalic = true;
            BackgroundBrush = Code.CreateBackgroundBrush();
            BackgroundOpacity = Code.GetBackgroundOpacity();
        }

    }

}
